posthog/api/test/test_dashboard.py

```diff
@@ -182,15 +184,67 @@ def test_return_cached_results(self):
         self.assertAlmostEqual(item.last_refresh, now(), delta=timezone.timedelta(seconds=5))
         self.assertEqual(item.filters_hash, generate_cache_key(f"{filter.toJSON()}_{self.team.pk}"))
 
-        with self.assertNumQueries(13):
-            # Django session, PostHog user, PostHog team, PostHog org membership, PostHog dashboard,
-            # PostHog dashboard item, PostHog team, PostHog dashboard item UPDATE, PostHog team,
-            # PostHog dashboard item UPDATE, PostHog dashboard UPDATE, PostHog dashboard item, Posthog org tags
-            response = self.client.get(f"/api/projects/{self.team.id}/dashboards/%s/" % dashboard.pk).json()
+        response = self.client.get(f"/api/projects/{self.team.id}/dashboards/%s/" % dashboard.pk).json()
 
         self.assertAlmostEqual(Dashboard.objects.get().last_accessed_at, now(), delta=timezone.timedelta(seconds=5))
         self.assertEqual(response["items"][0]["result"][0]["count"], 0)
 
+    def _get_dashboard_counting_queries(self, dashboard_id: int) -> Tuple[int, List[Dict[str, str]]]:
+        db_connection = connections[DEFAULT_DB_ALIAS]
+
+        with CaptureQueriesContext(db_connection) as capture_query_context:
+            response = self.client.get(f"/api/projects/{self.team.id}/dashboards/{dashboard_id}/")
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            query_count = len(capture_query_context.captured_queries)
+            return query_count, capture_query_context.captured_queries
+
+    def test_adding_insights_is_not_nplus1_for_gets(self):
+        dashboard_id, _ = self._create_dashboard({"name": "dashboard"})
+
+        filter_dict = {
+            "events": [{"id": "$pageview"}],
+            "properties": [{"key": "$browser", "value": "Mac OS X"}],
+            "insight": "TRENDS",
+        }
+        filter = Filter(data=filter_dict)
+
+        query_counts: List[int] = []
+        queries: List[List[Dict[str, str]]] = []
+
+        count, qs = self._get_dashboard_counting_queries(dashboard_id)
+        query_counts.append(count)
+        queries.append(qs)
+
+        # add insights to the dashboard and count how manh queries to read the dashboard afterwards
```

Comment:
What's the significance of adding four insights? If we are checking for O(1) queries then we only need to check two datapoints?

Reply:
It helped me see how quickly it climbed...

e.g. difference between 2, 4, 6, 8, and 2, 20, 38, 56